examples/webgl_tiled_forward.html

```diff
@@ -0,0 +1,343 @@
+		<script src="js/postprocessing/RenderPass.js"></script>
+		<script src="js/postprocessing/MaskPass.js"></script>
+		<script src="js/postprocessing/ShaderPass.js"></script>
+		<script src="js/shaders/CopyShader.js"></script>
+		<script src="js/shaders/ConvolutionShader.js"></script>
+		<script src="js/shaders/LuminosityHighPassShader.js"></script>
+		<script src="js/postprocessing/UnrealBloomPass.js"></script>
+		<script src="js/Detector.js"></script>
+		<script src="js/libs/stats.min.js"></script>
+
+		<script>
+		
+		// Simple form of tiled forward lighting 
+		// using texels as bitmasks of 32 lights
+
+		var RADIUS = 75;
+
+		THREE.ShaderChunk["lights_pars"] += [
+			"#if defined TILED_FORWARD",
+			"uniform vec4 tileData;",
+			"uniform sampler2D tileTexture;",
+			"uniform sampler2D lightTexture;",
+			"#endif"
+		].join("\n");
+
+		THREE.ShaderChunk["lights_template"] += [
+			"",
+			"#if defined TILED_FORWARD",
+			"vec2 tUv = floor(gl_FragCoord.xy / tileData.xy * 32.) / 32. + tileData.zw;",
+			"vec4 tile = texture2D(tileTexture, tUv);",
+			"for (int i=0; i < 4; i++) {",
+			"	float tileVal = tile.x * 255.;",	
+			"  	tile.xyzw = tile.yzwx;",
+			"	if(tileVal == 0.){ continue; }",
+			"  	float tileDiv = 128.;",
+			"	for (int j=0; j < 8; j++) {",
+			"  		if (tileVal < tileDiv) {  tileDiv *= 0.5; continue; }",
+			"		tileVal -= tileDiv;",
+			"		tileDiv *= 0.5;",
+			"  		PointLight pointlight;",
+			"		float uvx = (float(8 * i + j) + 0.5) / 32.;",
+			"  		vec4 lightData = texture2D(lightTexture, vec2(uvx, 0.));",
+			"  		vec4 lightColor = texture2D(lightTexture, vec2(uvx, 1.));",
+			"  		pointlight.position = lightData.xyz;",
+			"  		pointlight.distance = lightData.w;",
+			"  		pointlight.color = lightColor.rgb;",
+			"  		pointlight.decay = lightColor.a;",
+			"  		getPointDirectLightIrradiance( pointlight, geometry, directLight );",
+			"		RE_Direct( directLight, geometry, material, reflectedLight );",
+			"	}",
+			"}",
+			"#endif"
+		].join("\n");
+
+		var lights = [], objects = [];
+
+		var State = {
+			rows:0,
+			cols:0,
+			width: 0,
+			height: 0,
+			tileData : { type: "v4", value: null },
+			tileTexture: { type: "t", value: null },
+			lightTexture: {
+				type: "t", 
+				value: new THREE.DataTexture(new Float32Array(32 * 2 * 4), 32, 2, THREE.RGBAFormat, THREE.FloatType)
+			},
+		};
+
+		function resizeTiles() {
+			var width = window.innerWidth;
+			var height = window.innerHeight;
+			
+			State.width = width;
+			State.height = height;
+			State.cols = Math.ceil(width / 32);
+			State.rows = Math.ceil(height / 32);
+			State.tileData.value = [ width, height, 0.5 / Math.ceil( width / 32), 0.5 / Math.ceil( height / 32) ];
+			State.tileTexture.value = new THREE.DataTexture( new Uint8Array(State.cols * State.rows * 4), State.cols, State.rows);
+		}
+
+		// Generate the light bitmasks and store them in the tile texture
+		function tileLights(renderer, scene, camera) {
```

Comment:
Can you clean up your whitespace formatting a bit? https://zz85.github.io/mrdoobapproves/